upstairs/src/test.rs

```diff
@@ -711,27 +700,32 @@ pub(crate) mod up_test {
         Ok(())
     }
 
-    macro_rules! assert_consumed {
-        ($guest:expr) => {{
-            let req = $guest.consume_req().await;
-            assert!(req.is_some());
-            // Send a completion for the req, rather than just dropping it.
-            // This fulfills the invariant that BlockRes is completed prior to
-            // being dropped.
-            req.unwrap().res.send_ok();
-        }};
+    async fn assert_consumed(io: &mut GuestIoHandle) {
+        tokio::select! {
+            _ = io.recv() => {
+                // correct!
+            },
+            _ = tokio::time::sleep(std::time::Duration::from_millis(1)) => {
+                panic!("timed out while waiting for message");
+            }
+        }
     }
-    macro_rules! assert_none_consumed {
-        ($guest:expr) => {{
-            assert!($guest.consume_req().await.is_none());
-        }};
+
+    async fn assert_none_consumed(io: &mut GuestIoHandle) {
+        tokio::select! {
+            _ = io.recv() => {
+                panic!("got message when expecting nothing")
+            },
+            _ = tokio::time::sleep(std::time::Duration::from_millis(1)) => {
```

Comment:
Whats the purpose of the sleep here (and, really the above one also)?

Reply:
`io.recv()` will either
1) return an item (if available)
2) wait for an available item (if nothing is available but we're not BW / IOP limited), or
3) wait forever (if we're BW / IOP limited)

This makes sense in production use, because it's always used as part of a `select!` statement (so if it waits forever, a timeout branch in the `select!` will be taken).  In the test suite, we're always either doing (1) or (3), so this `select!` distinguishes between the "return right away" and "wait forever" cases.